apis/python/src/tiledbsc/soma.py

```diff
@@ -354,20 +336,82 @@ def write_annotation_matrices(self, annotation_matrices, name: str):
             # scipy.sparse.csr.csr_matrix. For ongoing work we will likely need more checks
             # here. See also desc-ann.py in this directory which helps reveal the datatypes
             # contained within a given HDF5 file.
-            input_as_np_array = annotation_matrices[name]
-            if isinstance(input_as_np_array, scipy.sparse.csr.csr_matrix):
-                input_as_np_array = input_as_np_array.toarray()
+            if name in ["obsm", "varm"]:
+                if isinstance(mat, scipy.sparse.csr_matrix):
+                    mat = mat.toarray()
```

Comment:
If this is csr do we still want `toarray()` here?

Reply:
No but I didn't touch `obsm`/`varm` in this PR. They need to be handled slightly difference since its 1D data and @johnkerl already started in a separate branch.